lib/spack/spack/ci/__init__.py

```diff
@@ -581,22 +582,26 @@ def copy_stage_logs_to_artifacts(job_spec: spack.spec.Spec, job_log_dir: str) ->
     tty.debug(f"job spec: {job_spec}")
 
     try:
-        pkg_cls = spack.repo.PATH.get_pkg_class(job_spec.name)
-        job_pkg = pkg_cls(job_spec)
-        tty.debug(f"job package: {job_pkg}")
+        package_metadata_root = pathlib.Path(spack.store.STORE.layout.metadata_path(job_spec))
     except AssertionError:
```

Comment:
If there are no more asserts, should this be removed?  Looking at the [function](https://github.com/spack/spack/blob/e2c6914dfe9741be032070f415ca5a28dd5f3a08/lib/spack/spack/directory_layout.py#L217) called there, it doesn't seem it will raise at all, and does not care whether the spec is concrete. But maybe I'm missing something.

Reply:
If you follow the call path it eventually ends up here https://github.com/spack/spack/blob/develop/lib/spack/spack/spec.py#L2126

I should actually change that to a `spack.error.SpackError` for the error I am looking for here. I just assumed it would still try to get `Spec.package` but this uses the database since it is installed.

Reply:
Ah yeah, by now I should know better than to assume accessing a property can't raise.